content for Golang structs (#1898)
kamranahmedse/developer-roadmap: content/roadmaps/109-golang/content/100-go-basics/116-structs.md

> # Structs
> 
> Structs are user-defined types and they help us create a collection of data that describe a single entity.
> 
> <ResourceGroupTitle>Free Content</ResourceGroupTitle>
> 
> <BadgeLink colorScheme='blue' badgeText='Official Website' href='https://go.dev/tour/moretypes/2'>Go Structs</BadgeLink>
> <BadgeLink badgeText='Read' href='https://gobyexample.com/structs'>Go by Example: Structs</BadgeLink>
> <BadgeLink badgeText='Watch' href='https://www.youtube.com/watch?v=NMTN543WVQY'>Structs in golang - (by Hitesh Choudhary
>  on YouTube)</BadgeLink>

Reviewer: ```suggestion
# Structs

Structs are user-defined types that help us create a collection of data describing a single entity.

<ResourceGroupTitle>Free Content</ResourceGroupTitle>
<BadgeLink colorScheme='blue' badgeText='Official Website' href='https://go.dev/tour/moretypes/2'>Go Structs</BadgeLink>
<BadgeLink badgeText='Read' href='https://gobyexample.com/structs'>Go by Example: Structs</BadgeLink>
<BadgeLink badgeText='Watch' href='https://www.youtube.com/watch?v=NMTN543WVQY'>Structs in Go</BadgeLink>
```
